.github/workflows/deploy-prod.yml

```diff
@@ -0,0 +1,64 @@
+name: 🌏 Production Deployment
+
+on:
+  push:
+    tags:
+      - "*"
+
+concurrency:
+  group: ${{ github.ref }}
+
+permissions:
+  contents: read
+  packages: write
+  id-token: write
+
+jobs:
+  check-tag-branch:
+    name: Check Tag Hash Against Master Branch Hash
+    runs-on: ${{ vars.RUNNER_OS }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Get Tag and Master Commit Hashes
+        id: check_hashes
+        run: |
+          # Get the commit hash associated with the tag
+          COMMIT_HASH_TAG=$(git show-ref --tags -d | grep ${GITHUB_REF#refs/tags/} | awk '{print $1}')
+
+          # Get the commit hash from the master branch
+          COMMIT_HASH_MASTER=$(git rev-parse origin/master)
+
+          # Compare the two commit hashes
+          if [ "$COMMIT_HASH_TAG" != "$COMMIT_HASH_MASTER" ]; then
+            echo "Error: Commit hashes do not match. Stopping the pipeline."
+            exit 1
+          fi
+
+  handle-docker-image:
+    name: 🐳 Handle Docker Image
+    needs: [check-tag-branch]
+    uses: ./.github/workflows/docker-image.yml
+    secrets: inherit
+
+  lint-test-security-checks:
+    name: 🧪 Lint, Test and Security Check
+    needs: [handle-docker-image, check-tag-branch]
+    uses: ./.github/workflows/ci-checks.yml
```

Comment:
Production deployment does not need localstack deployment

Reply:
prod is running all tests other than localstack plan and apply. its done by `is_deploy_localstack: false` flag 
https://github.com/CraftsmenLtd/BloodConnect/pull/148/files/343d6d93b1c119efae254b6afc358fa1c3b95a9a#diff-4ce342fb97e7d7feb8a578c91ba89fe4c27ddf16b4306909fe6d90220141913dR52